VirtualFinland.UserAPI/src/VirtualFinland.UsersAPI/Helpers/Services/DateOnlyJsonConverter.cs

```diff
@@ -0,0 +1,18 @@
+﻿using System.Globalization;
+using System.Text.Json;
+using System.Text.Json.Serialization;
+
+public class DateOnlyJsonConverter : JsonConverter<DateOnly>
+{
+    private const string Format = "yyyy-MM-dd";
+
+    public override DateOnly Read(ref Utf8JsonReader reader, Type typeToConvert, JsonSerializerOptions options)
+    {
+        return DateOnly.ParseExact(reader.GetString(), Format, CultureInfo.InvariantCulture);
```

Comment:
Tässä on mahdollista että `reader.getString()` palauttaa nullan ja `DateOnly.ParseExact` ei sellaista katso hyvällä.

`DateOfBirth` on nullable, eli kyseenalainen, ja varmaan on yleisestikin DateOnly-tyypit hieman sellaisia joten tämä helpperiluokka voisi tukea nullablea. 

Esim. jotenkin näin:

```
using System.Globalization;
using System.Text.Json;
using System.Text.Json.Serialization;

public class DateOnlyJsonConverter : JsonConverter<DateOnly?>
{
    private const string Format = "yyyy-MM-dd";

    public override DateOnly? Read(ref Utf8JsonReader reader, Type typeToConvert, JsonSerializerOptions options)
    {
        var readerValue = reader.GetString();
        if (readerValue == null)
        {
            return null;
        }
        return DateOnly.ParseExact(readerValue, Format, CultureInfo.InvariantCulture);
    }

    public override void Write(Utf8JsonWriter writer, DateOnly? value, JsonSerializerOptions options)
    {
        writer.WriteStringValue(value?.ToString(Format, CultureInfo.InvariantCulture));
    }
}
```

Reply:
Joo, tämä hyvä lisä 👍